cli/azd/pkg/azdo/utils.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+)
+
+// helper method to verify that a configuration exists in the .env file or in system environment variables
+func ensureConfigExists(ctx context.Context, env *environment.Environment, key string, label string) (string, error) {
+	value := env.Values[key]
+	if value != "" {
+		return value, nil
+	}
+
+	value, exists := os.LookupEnv(key)
+	if !exists || value == "" {
+		return value, fmt.Errorf("%s not found in environment variable %s", label, key)
+	}
+	return value, nil
+}
+
+// helper method to ensure an Azure DevOps PAT exists either in .env or system environment variables
+func EnsurePatExists(ctx context.Context, env *environment.Environment, console input.Console) (string, error) {
+	value, err := ensureConfigExists(ctx, env, AzDoPatName, "azure devops personal access token")
+	if err != nil {
+		console.Message(ctx, output.WithWarningFormat("You need an Azure DevOps Personal Access Token (PAT). Please create a PAT by following the instructions here https://aka.ms/azure-dev/azdo-pat"))
+		pat, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message:      "Personal Access Token (PAT):",
+			DefaultValue: "",
+		})
+		if err != nil {
+			return "", fmt.Errorf("asking for pat: %w", err)
+		}
+		// set the pat as an environment variable for this cmd run
+		// note: the scope of this env var is only this shell invocation and won't be available in the caller parent shell
+		os.Setenv(AzDoPatName, pat)
```

Comment:
Given that this only affects the current shell invocation, should we really store it as an environment variable instead of a variable in the program?

Reply:
This was requested by PM during last demo review